flutter/test/sentry_flutter_test.dart

```diff
@@ -1,128 +1,143 @@
-      _channel.setMockMethodCallHandler(
-        (MethodCall methodCall) async => <String, dynamic>{},
+    test('macOS', () async {
+      await SentryFlutter.init(
+        getConfigurationTester(
+          hasFileSystemTransport: true,
+          shouldHaveIntegrations: [
+            ...iOsAndMacOsIntegrations,
+            ...nativeIntegrations,
+            ...platformAgnosticIntegrations,
+          ],
+          shouldNotHaveIntegrations: androidIntegrations,
+        ),
+        appRunner: appRunner,
+        packageLoader: loadTestPackage,
+        platformChecker: getPlatformChecker(platform: MockPlatform.macOs()),
       );
-      when(transport.send(any))
-          .thenAnswer((realInvocation) => Future.value(SentryId.newId()));
     });
 
-    tearDown(() async {
-      _channel.setMockMethodCallHandler(null);
-      await Sentry.close();
-    });
-
-    test('should add loadContextsIntegration on ios', () async {
+    test('Windows', () async {
       await SentryFlutter.init(
-        (options) => options
-          ..dsn = fakeDsn
-          ..transport = transport,
+        getConfigurationTester(
+          hasFileSystemTransport: false,
+          shouldHaveIntegrations: platformAgnosticIntegrations,
+          shouldNotHaveIntegrations: [
+            ...androidIntegrations,
+            ...iOsAndMacOsIntegrations,
+            ...nativeIntegrations,
+          ],
+        ),
+        appRunner: appRunner,
         packageLoader: loadTestPackage,
-        channel: _channel,
-        platformChecker: getPlatformChecker(isIOS: true),
+        platformChecker: getPlatformChecker(platform: MockPlatform.windows()),
       );
-
-      await Sentry.captureMessage('a message');
-
-      final event =
-          verify(transport.send(captureAny)).captured.first as SentryEvent;
-
-      expect(event.sdk!.integrations.length, 7);
-      expect(event.sdk!.integrations.contains('loadContextsIntegration'), true);
     });
 
-    test('should not add loadContextsIntegration if not ios', () async {
+    test('Linux', () async {
       await SentryFlutter.init(
-        (options) => options
-          ..dsn = fakeDsn
-          ..transport = transport,
+        getConfigurationTester(
+          hasFileSystemTransport: false,
+          shouldHaveIntegrations: platformAgnosticIntegrations,
+          shouldNotHaveIntegrations: [
+            ...androidIntegrations,
+            ...iOsAndMacOsIntegrations,
+            ...nativeIntegrations,
+          ],
+        ),
+        appRunner: appRunner,
         packageLoader: loadTestPackage,
-        platformChecker: getPlatformChecker(isAndroid: true),
-        channel: _channel,
-      );
-
-      await Sentry.captureMessage('a message');
-
-      final event =
-          verify(transport.send(captureAny)).captured.first as SentryEvent;
-
-      expect(event.sdk!.integrations.length, 7);
-      expect(
-        event.sdk!.integrations.contains('loadContextsIntegration'),
-        false,
```

Comment:
this should be tested on load_contexts_integrations_test if not yet

Reply:
This is covered by the rewritten tests.